### PR TITLE
[for 2.6.0] Merge generic code of apps feature

### PR DESCRIPTION
**Pull request by Skylsmoi**
https://github.com/tracim/tracim/issues/1560

## Checkpoints

The goal of this section is to help code reviewer to do required checks on this pull request. Please don't edit this list.
If one or more of these checkpoints are out of scope, please write below why they are.

- [x] Code is clear enough
- [x] If there are FIXME in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODO, NOTE or HACK in code, date and developer initials are present
- [x] Automated tests have been written (covering feature or non regression), or *at least* an issue has been created for test implementation
- [x] Tested by quality team

### CODE REVIEW

**Comment by Skylsmoi** [comment]
Added new TODOS:
- [ ] put `appFeatureCustomEventHandlerShowApp` from `frontend_lib/src/helper.js` in appContentFactory.js
  - [ ] the function should use GLOBAL_dispatchEvent
- [ ] check to merge some code of custom event

File: frontend_lib/src/appContentFactory.js [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+    }
+
+    appContentChangeTitle = async (content, newTitle, appSlug, propertiesToAddToBody = {}) => {
+      this.checkApiUrl()
+
+      if (content.label === newTitle) return
+
+      // INFO - CH - 2019-01-03 - Check the -s added to the app slug. This is and should stay consistent with app features
+      const response = await handleFetchResult(
+        await fetch(`${this.apiUrl}/workspaces/${content.workspace_id}/${appSlug}s/${content.content_id}`, {
```
Comment (metzgegu) [inline frontend_lib/src/appContentFactory.js]: For a better consistency, maybe should we move `frontend_lib` api calls in a `src/action.async.js` file like in apps ?

Comment (Skylsmoi) [reply frontend_lib/src/appContentFactory.js]: I agree. I'll do the changes

File: frontend_app_folder_advanced/src/container/FolderAdvanced.jsx [hunk 2]
```diff
@@ -51,32 +49,31 @@ class FolderAdvanced extends React.Component {
   }
 
   customEventReducer = ({ detail: { type, data } }) => { // action: { type: '', data: {} }
```
Comment (metzgegu) [inline frontend_app_folder_advanced/src/container/FolderAdvanced.jsx]: Is this comment relevant ?

File: frontend_lib/test/appContentFactory.spec.js [hunk 3]
```diff
@@ -0,0 +1,565 @@
...
+        )
+      })
+    })
+  })
+
+  describe('function buildTimelineFromCommentAndRevision', () => {
+    const commentList = fixtureCommentList
+    const revisionList = fixtureRevisionList.map((revision, i) => ({
+      ...revision,
+      // ensure that the first revision after creation has all the comments from commentList
```
Comment (metzgegu) [inline frontend_lib/test/appContentFactory.spec.js]: Comments should respect the comment rules

**Comment by PhilippeAccorsi** [comment]
@Skylsmoi some warning at build
~~~
WARNING in ./src/container/HtmlDocument.jsx
Module Warning (from ./node_modules/standard-loader/index.js):
(Emitted value instead of an instance of Error) 
/home/philippe/repo/tracim/frontend_app_html-document/src/container/HtmlDocument.jsx
  20:3  error  'appFeatureCustomEventHandlerShowApp' is defined but never used  no-unused-vars

✖ 1 problem

 @ ./src/index.js 5:0-56 14:47-59
~~~
WARNING in ./src/container/Thread.jsx
Module Warning (from ./node_modules/standard-loader/index.js):
(Emitted value instead of an instance of Error) 
/home/philippe/repo/tracim/frontend_app_thread/src/container/Thread.jsx
  17:3  error  'appFeatureCustomEventHandlerShowApp' is defined but never used  no-unused-vars

✖ 1 problem

 @ ./src/index.js 5:0-44 14:47-53
~~~

~~~
WARNING in ./src/container/File.jsx
Module Warning (from ./node_modules/standard-loader/index.js):
(Emitted value instead of an instance of Error) 
/home/philippe/repo/tracim/frontend_app_file/src/container/File.jsx
  23:3  error  'appFeatureCustomEventHandlerShowApp' is defined but never used  no-unused-vars

✖ 1 problem

 @ ./src/index.js 5:0-40 14:47-51
~~~

**Comment by PhilippeAccorsi** [comment]
@Skylsmoi just fix warning and you can merge.
Tested on branch.

**Comment by Skylsmoi** [comment]
PR about refactor of ROLE and PROFILE merged, creating new conflicts

**Pull request merged**